Hublsoft.Net.LoginLogout.Bll/UserManager.cs

```diff
@@ -42,6 +42,14 @@ public async Task<Guid> AuthenticateUserAsync(string emailAddress, string passwo
                 tasks.Add(_registeredUsersRepo.IncrementFailedLoginAttemptsAsync(id));
                 tasks.Add(_registeredUserAuditsRepo.AddFailedLoginAttemptAuditRecordAsync(id));
                 await Task.WhenAll(tasks);
+
+                var loginAttempts = await _registeredUsersRepo.GetFailedLoginAttempsAsync(id);
+
+                if (loginAttempts == 3)
```

Comment:
There is potential here for simultaneous attempts to cause `loginAttempts` to be incremented to a number greater than 3 and therefore bypass the `LockAccount` method, leaving the system open to attack

Reply:
Oh yes that's a good spot. Making this loginAttempts > 2 would prevent this. I would also suggest we should move this logic into the record failed login method. That way it is all self contained and avoids any race conditions.

Reply:
The very next test scenario would be for the login process to check if the account is locked before proceeding any further. This should prevent the race condition from happening as well.